code/ZAS/Fire.dm

```diff
@@ -58,13 +58,21 @@ Attach to transfer valve and open. BOOM.
 	on_fire=0
 	if(fire_overlay)
 		overlays -= fire_overlay
+	if(istype(src,/obj))
+		var/obj/O = src
+		O.source_temperature = initial(O.source_temperature)
 
 /atom/proc/ignite(var/temperature)
 	on_fire=1
 	//visible_message("\The [src] bursts into flame!")
 	if(fire_dmi && fire_sprite)
 		fire_overlay = image(fire_dmi,fire_sprite)
 		overlays += fire_overlay
+	if(istype(src,/obj))
```

Comment:
>typechecking `src`

Don't do this. Put it in `/obj/ignite()`

Reply:
thanks, do i need a ..() in there?

Reply:
that's a yes